### PR TITLE
Gamma-weighted focal loss with pre-trained model

### PR DESCRIPTION
**Pull request by MattsonCam**
This PR replaces the previous L1-SSIM training objective with a gamma-weighted absolute-error loss. It also updates validation loss tracking so early stopping and model selection use the new objective consistently. In addition, it adds optional initialization from a logged MLflow model ID to support fine-tuning from an existing model. The model chosen in this pr was trained using the joint L1-SSIM loss and the data was robustly normalized using the pixel percentiles. Also, the learning rate now uses a narrower range for fine-tuning.

### CODE REVIEW

File: callbacks/utils/SaveEpochCrops.py [hunk 1]
```diff
@@ -83,6 +85,60 @@ def save_image(
...
+            [target_image.detach().reshape(-1), prediction_image.detach().reshape(-1)]
+        )
+        pair_np = pair_values.float().cpu().numpy()
+        lower = float(np.percentile(pair_np, 1.0))
+        upper = float(np.percentile(pair_np, 99.0))
+        if not np.isfinite(lower) or not np.isfinite(upper) or lower >= upper:
+            lower = float(np.min(pair_np))
+            upper = float(np.max(pair_np))
+        if lower >= upper:
+            upper = lower + 1.0
```
Comment (wli51) [inline callbacks/utils/SaveEpochCrops.py]: This can magnify the perceived level of noise. e.g.  if the prediction returns pure gaussian noise with low mean and low std or high mean and low std the resulting bound computed for both would be a mediocre gray. Maybe setting the lower bound to 0 whenever the internal contrast of image is too low would be a better fallback because you will be able to see whether the mean is actually high vs low.

File: train.py [hunk 2]
```diff
@@ -172,12 +326,14 @@ def __call__(self, trial: optuna.trial.Trial):
...
         }
 
-        loss_trainer = L1Loss()
-        loss_callbacks = L1(device=device)
+        loss_trainer = GammaWeightedL1Loss(gamma=gamma)
+        # Keep checkpoint selection aligned with the gamma-weighted training
+        # objective while denormalized image-quality metrics continue to be logged.
+        loss_callbacks = GammaWeightedL1LossMetric(gamma=gamma, device=device)
         metrics = [
             L2(device=device),
```
Comment (wli51) [inline train.py:336]: maybe the native l1 should be added to the metric as now the l1 loss is gamma weighted

**Review by wli51 — APPROVED** [review]
LGTM! Looking forward to see if and how focal loss improves prediction.

File: callbacks/utils/SaveEpochCrops.py (hunk 1 above)
Comment (MattsonCam) [reply callbacks/utils/SaveEpochCrops.py]: Good catch. I changed the target/prediction crop rendering to use the fixed denormalized target bounds instead of per-image percentile autoscaling

The displayed brightness now reflects the actual target-space intensity consistently across samples, so a flat low-mean prediction stays dark and a flat high-mean prediction stays bright. It also avoids stretching low-variance noise into something that looks more structured than it is

File: train.py (hunk 2 above)
Comment (MattsonCam) [reply train.py:336]: Good eye, I forgot to include this so I will do so
